### ISSUE TITLE
data.aws_ecs_service should return running task definition revision

### ISSUE THREAD

**Issue by joeig** [terraform-providers/terraform-provider-aws#1681]
I deploy new Docker containers to ECS using one task definition per container release (this is usually invoked by a CI job). Since `aws_ecs_service` requires a valid task definition name containing the revision ID, which is increased by the CI job, it is not possible to dynamically adapt the `aws_ecs_service` resource to the latest running task definition.

Wouldn't it be awesome if there was a data source for `aws_ecs_service` which is able to detect the running task definition revision?

```
data "aws_ecs_service" "running_voting" {
  cluster_name = "mycluster"
  service_name = "voting"
}

resource "aws_ecs_service" "voting" {
  name            = "voting"
  task_definition = "${data.aws_ecs_service.running_voting.arn}"
  desired_count   = 3
}
```

Terraform v0.10.4

**Comment by joeig** [comment]
One possible solution is to use `aws_ecs_task_definition` for that, but this just returns the latest active revision and not the actually running revision.

**Comment by bflad** [comment]
The new `aws_ecs_service` data source has been merged and will release with version 1.22.0 of the AWS provider, likely sometime middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.22.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1220-june-05-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by n1te1337** [comment]
@joeig isn't this essentially creating a circular dependency and prohibits the service from being created in the first place?
